frontend/src/components/Table/TablePagination.tsx

```diff
@@ -54,6 +54,7 @@ const TablePagination = <T extends object>(props: TablePaginationProps<T>): Reac
       marginPagesDisplayed={2}
       pageRangeDisplayed={5}
       onPageChange={handleChangePage}
+      forcePage={pageIndex}
```

Comment:
This ensures that the pagination control stays in sync with the internal state within react table.